exporter/loggingexporter/logging_exporter.go

```diff
@@ -25,18 +28,128 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
 )
 
+type traceDataBuffer struct {
+	str strings.Builder
+}
+
+func (b *traceDataBuffer) logEntry(format string, a ...interface{}) {
+	b.str.WriteString(fmt.Sprintf(format, a...))
+	b.str.WriteString("\n")
+}
+
+func (b *traceDataBuffer) logAttr(label string, value string) {
+	b.logEntry("    %-12s: %s", label, value)
+}
+
+func (b *traceDataBuffer) logMap(label string, data *map[string]string) {
+	if data == nil {
+		return
+	}
+
+	b.logEntry("%s:", label)
+	for label, value := range *data {
+		b.logEntry("     -> %s: %s", label, value)
+	}
+}
+
+type loggingExporter struct {
+	level  string
+	logger *zap.Logger
+}
+
+func (s *loggingExporter) pushTraceData(
+	ctx context.Context,
+	td consumerdata.TraceData,
+) (int, error) {
+	debug := s.level == "debug"
```

Comment:
Nit: better to do this once during creation and not for every `pushTraceData` call.